db/fetcher/fetcher.go

```diff
@@ -27,6 +27,28 @@ import (
 	"github.com/sourcenetwork/defradb/request/graphql/parser"
 )
 
+// Stats contains statistics about the fetcher execution.
+type Stats struct {
+	// Number of documents fetched.
+	DocsFetched uint64
+	// Number of fields fetched.
+	FieldsFetched uint64
+}
+
+// Add adds the other stats to the current stats.
+func (s Stats) Add(other Stats) Stats {
+	return Stats{
+		DocsFetched:   s.DocsFetched + other.DocsFetched,
+		FieldsFetched: s.FieldsFetched + other.FieldsFetched,
+	}
+}
```

Comment:
todo: Can just mutate `s`, similar to `Reset()`. Then update the usage in other places, as you are reassigning back to `s` anyway in all your usages.

```suggestion
func (s *Stats) Add(other Stats) {
	s.DocsFetched = s.DocsFetched + other.DocsFetched
	s.FieldsFetched = s.FieldsFetched + other.FieldsFetched
}
```